app/PccSyncManager.php

```diff
@@ -113,17 +122,173 @@ private function createOrUpdatePost($postId, Article $article, bool $isDraft = f
 			'post_name' => $article->slug,
 			'post_type' => $this->getIntegrationPostType(),
 		];
+		// Set post excerpt if description is available.
+		if (isset($article->metadata['description'])) {
+			$data['post_excerpt'] = $article->metadata['description'];
+		}
+
 		if (!$postId) {
 			$postId = wp_insert_post($data);
 			update_post_meta($postId, PCC_CONTENT_META_KEY, $article->id);
+			$this->syncPostMetaAndTags($postId, $article);
 			return $postId;
 		}
 
 		$data['ID'] = $postId;
 		wp_update_post($data);
+		$this->syncPostMetaAndTags($postId, $article);
 		return $postId;
 	}
 
+	/**
+	 * Update post tags.
+	 *
+	 * @param $postId
+	 * @param Article $article
+	 */
+	private function syncPostMetaAndTags($postId, Article $article): void
+	{
+		if (isset($article->tags) && is_array($article->tags)) {
+			wp_set_post_terms($postId, $article->tags, 'post_tag', false);
+		}
+
+		if (!isset($article->metadata)) {
+			return;
+		}
+
+		$this->setPostFeatureImage($postId, $article);
+		if (isset($article->metadata['Categories'])) {
+			wp_set_post_categories($postId, $this->findArticleCategories($article));
+		}
+
+		// Check if Yoast SEO is installed and active.
+		$activePlugins = apply_filters('active_plugins', get_option('active_plugins'));
+		if (in_array('wordpress-seo/wp-seo.php', $activePlugins)) {
+			if (isset($article->metadata['title'])) {
+				update_post_meta($postId, '_yoast_wpseo_title', $article->metadata['title']);
+			}
+			if (isset($article->metadata['description'])) {
+				update_post_meta($postId, '_yoast_wpseo_metadesc', $article->metadata['description']);
+			}
+		}
+	}
+
+	/**
+	 * Set the post feature image.
+	 *
+	 * @param $postId
+	 * @param Article $article
+	 */
+	private function setPostFeatureImage($postId, Article $article)
+	{
+		if (!isset($article->metadata['FeaturedImage'])) {
+			return;
+		}
+		// If the feature image is empty, delete the existing thumbnail.
+		$featuredImageURL = $article->metadata['FeaturedImage'] ?? '';
+		if ('' === $featuredImageURL) {
+			delete_post_thumbnail($postId);
+			return;
+		}
```

Comment:
remove these lines;
we don't want to remove the featured image from the WP article if the featured image is missing from the PCC article metadata

Reply:
As discussed, We will keep it as it is and clean up code